packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -498,4 +500,99 @@ export class ProjectModel {
 
         return cachedWarehouse;
     }
+
+    async getProjectAccess(
+        projectUuid: string,
+    ): Promise<ProjectMemberProfile[]> {
+        type QueryResult = {
+            user_uuid: string;
+            email: string;
+            role: ProjectMemberRole;
+            first_name: string;
+            last_name: string;
+        };
+        const projectMemberships = await this.database('project_memberships')
+            .leftJoin('users', 'project_memberships.user_id', 'users.user_id')
+            .leftJoin('emails', 'emails.user_id', 'users.user_id')
+            .leftJoin(
+                'projects',
+                'project_memberships.project_id',
+                'projects.project_id',
+            )
+            .select<QueryResult[]>()
+            .where('project_uuid', projectUuid);
+
+        return projectMemberships.map((membership) => ({
+            userUuid: membership.user_uuid,
+            email: membership.email,
+            role: membership.role,
+            firstName: membership.first_name,
+            projectUuid,
+            lastName: membership.last_name,
+        }));
+    }
+
+    async createProjectAccess(
+        projectUuid: string,
+        email: string,
+        role: ProjectMemberRole,
+    ): Promise<void> {
+        const [project] = await this.database('projects')
+            .select('project_id')
+            .where('project_uuid', projectUuid);
+
+        const [user] = await this.database('users')
+            .leftJoin('emails', 'emails.user_id', 'users.user_id')
+            .select('users.user_id')
+            .where('email', email);
+        if (user === undefined) {
+            throw new NotExistsError(
+                `Can't find user with email ${email} in the organization`,
+            );
+        }
+
+        await this.database('project_memberships').insert({
+            project_id: project.project_id,
+            role,
+            user_id: user.user_id,
+        });
+    }
+
+    async updateProjectAccess(
+        projectUuid: string,
+        userUuid: string,
+        role: ProjectMemberRole,
+    ): Promise<void> {
+        const [project] = await this.database('projects')
```

Comment:
If you want you can do this whole update in one query to avoid 3 round trips to the DB, see `updateOrganizationMember`

Reply:
Nice, that is a lot more efficient, do you prefer to use `raw` in this case ?

Reply:
I couldn't find a way to do it with `knex` but it could be possible!

Reply:
I don't mind either way tho